src/main/java/uk/gov/pay/directdebit/payments/model/Transaction.java

```diff
@@ -15,17 +15,30 @@
     private String paymentRequestDescription;
     private String paymentRequestReturnUrl;
     private String paymentRequestReference;
-    private Long paymentRequestGatewayAccountId;
+    private Long gatewayAccountId;
+    private String gatewayAccountExternalId;
     private PaymentProvider paymentProvider;
     private Long amount;
     private Type type;
     private PaymentState state;
 
-    public Transaction(Long id, Long paymentRequestId, String paymentRequestExternalId, String paymentRequestDescription, String paymentRequestReference, Long paymentRequestGatewayAccountId, PaymentProvider paymentProvider, String paymentRequestReturnUrl, Long amount, Type type, PaymentState state) {
+    public Transaction(Long id,
+                       Long paymentRequestId,
+                       String paymentRequestExternalId,
```

Comment:
This is what I am talking about. I see Transaction having an entire PaymentRequest object and potentially a GatewayAccount in its constructor. But again, it's ok to leave it as it is, for now.